### PR TITLE
Add Ctrlz support to Pilot.

### PR DESCRIPTION
**Pull request by ozevren**
This is a straight-up integration of CtrlZ to Pilot.

### CODE REVIEW

**Comment by ozevren** [comment]
@costinm,

I tried looking at envoy/v2/ads.go but couldn't figure out what you think should be changed. Let me know and I can add.

Also, I  considered using the monitoring port as the default, but wouldn't it conflict? It looks like there is a listener being initialized already on that port. I opted to do the simplest integration possible.

File: pilot/cmd/pilot-agent/main.go [hunk 1]
```diff
@@ -67,6 +68,8 @@ var (
 
 	loggingOptions = log.DefaultOptions()
 
+	ctrlzOptions = ctrlz.DefaultOptions()
```
Comment (costinm) [inline pilot/cmd/pilot-agent/main.go]: I'm not sure about agent opening an extra port and exposing this.

File: pilot/pkg/bootstrap/server.go [hunk 2]
```diff
@@ -214,6 +216,8 @@ func NewServer(args PilotArgs) (*Server, error) {
 		return nil, err
 	}
 
+	go ctrlz.Run(args.CtrlZOptions, nil)
```
Comment (costinm) [inline pilot/pkg/bootstrap/server.go]: Same here - no need in agent (AFAIK)

**Comment by costinm** [comment]
I'll make the changes to ads ( to use the new logging ). But please remove ctlrz from agent and bootstrap.

**Comment by costinm** [comment]
And it should go to 0.8 I think. We'll cut another release candidate probably - it's ok if it doesn't make it in today build. It adds a lot more visibility into bad state/settings in pilot, I would like testers to have access to it.

**Comment by ozevren** [comment]
I already removed it from agent. If I remove it from bootstrap, then I won't be able to attach configz stuff easily. That taps into the cached store.

If you have a fundamental objection to keeping the code in bootstrap, I can pass a lambda (or something similar) as part of args/NewServer call to wire things up. Otherwise, it is simpler to keep ctrlz startup in bootstrap.

**Comment by ozevren** [comment]
I can do a cherry-pick to 0.8 once this lands, btw.

**Comment by ldemailly** [comment]
> I can do a cherry-pick to 0.8 once this lands, btw.

it's probably best/easier to just click edit at the top of this PR and change the base to be release-0.8 - most development should happen in that branch and will get mass merged back to master regularly (like #5590 )

**Pull request closed without merging**